src/pages/en/reference/integrations-reference.md

```diff
@@ -40,30 +39,159 @@ interface AstroIntegration {
+
+A callback function to add a component framework renderer (i.e. React, Vue, Svelte, etc). You can browse the examples and type definition above for more advanced options, but here are the 2 main options to be aware of:
+- `clientEntrypoint` - path to a file that executes on the client whenever your component is used. This is mainly for rendering or hydrating your component with JS.
+- `serverEntrypoint` - path to a file that executes during server-side requests or static builds whenever your component is used. These should render components to static markup, with hooks for hydration where applicable. [React's `renderToString` callback](https://reactjs.org/docs/react-dom-server.html#rendertostring) is a classic example.
+
+#### "injectScript" option
+
+**Type:** `(stage: InjectedScriptStage, content: string) => void;`
+
+A callback function to inject a string of JavaScript content onto every page.
+
+The **`stage`** denotes how this script (the `content`) should be inserted. Some stages are inserted purely, while others are picked up by Vite:
+
+- `"before-hydration"`: Imported client-side, before the hydration script runs. Processed & resolved by Vite.
+- `"head-inline"`: Injected into a script tag in the `<head>` of every page. Not processed or resolved by Vite.
+- `"page"`: Injected into the JavaScript bundle of every page. Processed & resolved by Vite.
+- `"page-ssr"`: Injected into the frontmatter of every Astro page. Processed & resolved by Vite.
+
 ### astro:config:done
+
+**When it's run:** after the Astro config has resolved and other integrations have run their `astro:config:setup` hooks.
+
+**Use case:** to retrieve the final config for use in other hooks.
+
+#### "config" option
+
+**Type**: `AstroConfig`
+
+A read-only copy of the user-supplied [Astro config](/en/reference/configuration-reference/). This is resolved _after_ other integrations have run.
+
 ### astro:server:setup
+
+**When it's run:** just after the Vite server is created in "dev" or "preview" mode, but before the `listen()` event is fired. [See Vite's createServer API](https://vitejs.dev/guide/api-javascript.html#createserver) for more.
+
+**Use case:** to update Vite server options and middleware.
+
+#### "server" option
+
+**Type:** [`ViteDevServer`](https://vitejs.dev/guide/api-javascript.html#vitedevserver)
+
+An mutate-able instance of the Vite server used in "dev" and "preview" mode. For instance, this is [used by our Partytown integration](https://github.com/withastro/astro/tree/main/packages/integrations/partytown) to inject the Partytown server as middleware:
+
+```js
+'astro:server:setup': ({ server }) => {
+  server.middlewares.use(
+    sirv(partytownLibDirectory, {
```

Comment:
just checking: serv?

Reply:
I can throw the `sirv` import at the top of the example for clarity, but I assumed it was a non-important detail 🤷

Reply:
Update: changing `sirv` to `partytownServer` to avoid confusion 👍